src/coreclr/tools/aot/ILCompiler/ConfigurablePInvokePolicy.cs

```diff
@@ -84,15 +84,17 @@ private IEnumerable<string> ModuleNameVariations(string name)
 
             if (_target.IsWindows)
             {
-                if (name.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
-                    yield return name.Substring(0, name.Length - 4);
+                string suffix = ".dll";
+
+                if (name.EndsWith(suffix, StringComparison.OrdinalIgnoreCase))
+                    yield return name.Substring(0, name.Length - suffix.Length);
             }
             else
             {
                 string suffix = _target.IsOSX ? ".dylib" : ".so";
 
                 if (name.EndsWith(suffix, StringComparison.Ordinal))
```

Comment:
macOS is not case-sensitive so I think the suffix comparasion logic should be the same with Windows for OSX.

Reply:
CoreCLR uses case-sensitive comparisons for these as well, e.g.  https://github.com/dotnet/runtime/blob/e98fb61d8b4bb6687ccddead861d8b140751647b/src/coreclr/vm/dllimport.cpp#L6062

since there is a lof